addons/osio-addon/src/main/java/io/fabric8/launcher/osio/client/AnalyticsClient.java

```diff
@@ -0,0 +1,79 @@
+package io.fabric8.launcher.osio.client;
+
+
+import java.io.IOException;
+
+
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+
+import io.fabric8.launcher.base.EnvironmentSupport;
+import io.fabric8.launcher.base.http.HttpClient;
+import io.fabric8.launcher.base.http.HttpException;
+import io.fabric8.launcher.base.identity.TokenIdentity;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
+import static io.fabric8.launcher.base.http.Requests.securedRequest;
+
+import static io.fabric8.utils.URLUtils.pathJoin;
+import static java.util.Objects.requireNonNull;
+
+@RequestScoped
+public class AnalyticsClient {
+
+
+    private final TokenIdentity authorization;
+
+    private final HttpClient httpClient;
+    private static final String analyticsUrl = EnvironmentSupport.getRequiredEnvVarOrSysProp("F8A_ANALYTICS_RECOMMENDER_API_URL");
+
+    @Inject
+    public AnalyticsClient(final TokenIdentity authorization, HttpClient httpClient) {
+        this.authorization = requireNonNull(authorization, "authorization must be specified.");
+        this.httpClient = requireNonNull(httpClient, "httpClient must be specified");
+    }
+
+    /**
+     * no-args constructor used by CDI for proxying only
+     * but is subsequently replaced with an instance
+     * created using the above constructor.
+     *
+     * @deprecated do not use this constructor
+     */
+    @Deprecated
+    protected AnalyticsClient() {
+        this.authorization = null;
+        this.httpClient = null;
+    }
+
+    public Response analyticsRequest(String path, RequestBody body) {
+        final Request request = newAuthorizedRequestBuilder(path)
+                .post(body)
+                .build();
+        Response response = null;
+
+        try {
+            response = httpClient.getClient().newCall(request).execute();
```

Comment:
Use the `HttpClient.executeAndMap` here

Reply:
Actually just `HttpClient.execute` https://github.com/fabric8-launcher/launcher-backend/blob/master/base/src/main/java/io/fabric8/launcher/base/http/HttpClient.java#L89

Reply:
didn't get it.

Reply:
Earlier, I was doing the same but then I wasn't able to assert the response https://github.com/ravsa/launcher-backend/blob/d0444aaa5e59c918aa81cda55ffcc1e044543096/addons/osio-addon/src/test/java/io/fabric8/launcher/osio/client/AnalyticsClientTest.java#L49

Reply:
Ah so it returns a JSON? Then I think you can use HttpClient.executeAndParseJson

Reply:
Can you create a representation of the response as an interface with Immutable annotations? You can use that in your test instead of dealing with JSON directly

Reply:
Actually, JSON structure is dynamic.

Reply:
If it's dynamic then I don't think it makes sense to test the JSON result in the test case.

Reply:
then I think I should check for the successful status.